### PR TITLE
Change default changefreq from always to monthly

### PR DESCRIPTION
**Pull request by dan-jensen**
### What does this do?

This changes the default changefreq from "always" to "monthly" to avoid
causing Google Search Console errors, and to be a better default in
general, because "always" is rarely appropriate and "monthly"
frequently is.

This also unifies the 2 uses of the default changefreq into a single
configuration variable. In the future this configuration variable
could become officially supported.

### Background

- Currently, `sitemap_generator` uses "always" as the default changefreq.
- [Google sitemap protocol says](https://www.google.com/sitemaps/protocol.html): "The value "always" should be used to describe documents that change each time they are accessed."

### Problems to solve

1. Documents rarely change each time they are accessed, so "always" is probably not a good default value.
1. Google Search Console will raise an "Invalid tag value" error when a
sitemap entry has a changefreq of "always" but is rarely changed. (This
is presumably because Google does not want to be encouraged to crawl a
page unnecessarily, as that is a waste of resources.)
1. There is no way to alter the changefreq of auto-generated/default sitemap entries ("SitemapGenerator adds the root URL / for you by default.")

### CODE REVIEW

File: lib/sitemap_generator.rb [hunk 1]
```diff
@@ -54,9 +54,10 @@ def reset!
   end
 
   class << self
-    attr_accessor :root, :app, :templates
+    attr_accessor :default_changefreq, :root, :app, :templates
     attr_writer :yield_sitemap, :verbose
   end
+  @default_changefreq = 'monthly'
```
Comment (dan-jensen) [inline lib/sitemap_generator.rb:60]: Would `weekly` be a better default?  I'm not certain `monthly` is best, but I'm also not certain it matters very much 🤷

**Comment by dan-jensen** [comment]
@kjvarga ready for your review 🙂

**Comment by kjvarga** [comment]
Thanks @dan-jensen this makes sense.  But I'm thinking to skip adding the configuration attribute and just alter the defaults for the index and root links.  Then if someone wants to use a different value they can just turn off `include_root` and add the root entry manually and specify their desired `changefreq`, which is trivial.

**Comment by dan-jensen** [comment]
@kjvarga okay, I just replaced the configuration attribute with a constant.  How's this looking now?

**Review by kjvarga — APPROVED** [review]

**Comment by kjvarga** [comment]
@dan-jensen I found I could simplify the code a bit by allowing the default on the SitemapUrl class to be used.  And that meant the default const was no longer needed.  Thanks for the code contribution!

**Comment by kjvarga** [comment]
Released in 6.2.0

**Comment by naman-modi** [comment]
Hey @kjvarga, any way we can choose to not have any `changefreq`, and `priority` tags?

Would be helpful, thanks!

**Pull request merged**